lib/tpm2_alg_util.c

```diff
@@ -330,3 +330,57 @@ bool pcr_parse_digest_list(char **argv, int len,
 
     return true;
 }
+
+UINT8* tpm2_extract_plain_signature(UINT16 *size, TPMT_SIGNATURE *signature) {
+
+    UINT8 *buffer = NULL;
+    *size = 0;
+
+    switch (signature->sigAlg) {
+    case TPM_ALG_RSASSA:
+        *size = sizeof((*signature).signature.rsassa.sig.t.buffer);
+        buffer = malloc(*size);
+        if(! buffer) {
+            goto nomem;
+        }
+        memcpy(buffer, (*signature).signature.rsassa.sig.t.buffer, *size);
+        break;
+    case TPM_ALG_HMAC: {
+        TPMU_HA *hmac_sig = &((*signature).signature.hmac.digest);
+        *size = tpm2_alg_util_get_hash_size((*signature).signature.hmac.hashAlg);
+        buffer = malloc(*size);
+        if(! buffer) {
+            goto nomem;
+        }
+        memcpy(buffer, &(hmac_sig->na), *size);
```

Comment:
No need for the parenthesis here, the arrow operator has a higher precedence than the address-of operator.